### ISSUE TITLE
Remove User Acount

### ISSUE THREAD

**Issue by Twilek-de** [sdelements/lets-chat#256]
Is there any way yet to remove or reset a user account with a lost password without directly hacking the database?

**Comment by sibartlett** [comment]
There currently isn't a way to remove or reset a lost password.

We have plans for both, with the latter being of very high priority.

**Comment by geoffwhittington** [comment]
When deleting a user I assume the user is an employee and its employer normally has reasonable rights to retain the user's data. The following points come to mind:

- User-contributed data - I don't think we need to wipe anything
- User record - Mark it deleted and disallow further authentication attempts by the user
- Rooms owned by the user - Archive their rooms automatically (bonus: enable ownership transfer)
- Existing room membership - Remove them from all rooms
- Transcripts, file lists, etc - display data normally but note that the user is deleted

**Comment by sibartlett** [comment]
@geoffwhittington yes, you're quite right; we'd want to disable the user rather than delete them

**Comment by Twilek-de** [comment]
I am not a legal expert, but as far as I know local law in Germany gives everyone the right to full disclosure of their data at any time and explicitly gives them the right to have their data wiped completely if they want it (which gave facebook quite a bit of a headache dealing with german authorities). I guess as long as lets chat is still a rather small project that legal stuff is quite academic but if the scope and the number of installations increase (as they are wont to do given the quality of lets chat ;-) ) it might become a problem one day. And of course resetting PWs is a must (you might also consider having the user to repeat the PW in the registration process as I had the problem with a user not being able to login after the registration probably because of a mistyped PW).

**Comment by JohnTheodore** [comment]
If people want to use this app in Germany and comply with all local laws, they can send a PR or fork. :p. If you took into account all the local laws for the various countries, nothing would get done.
